client-side/src/components/signUp/signUp.jsx

```diff
@@ -0,0 +1,149 @@
+    .email(MessagesSignUp.email.invalid)
+    .required(MessagesSignUp.email.required),
+  password: Yup.string()
+    .required(MessagesSignUp.password.required)
+    .matches(/[A-Za-z]/, MessagesSignUp.password.matches.letters)
+    .matches(/\d/, MessagesSignUp.password.matches.digits)
+    .min(4, MessagesSignUp.password.min)
+});
+function SignUp({ isEditMode = false, onSave }) {
+  const [password, setPassword] = useState('');
+//   const [robotPass, setRobotPass] = useState(null)
+  const url = process.env.REACT_APP_SITEKEY;
+  const dispatch = useDispatch();
+  const navigate=useNavigate();
+const user = useSelector(state => state.user.currentUser|| {});
+const userId =user.id || null;
+const formik = useFormik({
+    initialValues:{ name: user.name || '', email: user.email || '', password: '' },
+    validationSchema: SignUpSchema,
+   onSubmit: (values) => {
+    const user = {
+      name: values.name,
+      email: values.email,
+      password: values.password,
+    };
+    if (isEditMode) {
+      editUser(user);
+    } else {
+      userSignUp(user);
+    }
+  },
+  });
+useEffect(() => {
+  if (isEditMode) {
+    formik.setValues({ name: user.name || '', email: user.email || '', password: '' });
+  }
+}, [isEditMode]);
+
+  const userSignUp = async (user) => {
+    try {
+      await createUser(user); 
+      dispatch(addUser(user));
+      navigate('/home');
+    } catch (error) {
+      console.error("The user is not included in the system");
+    }
+  };
+  const editUser = async (user) => {
+    try {
+      await updateUser(user,userId); 
+      dispatch(updateUserDetails(user));
+      navigate('/home');
+    } catch (error) {
+      console.error("Failed to update user details");
+    }
+  };
+  return (
+    <div className="signup-container">
+    <h2>{isEditMode ? 'Edit User' : 'Sign Up'}</h2>
+      <form onSubmit={formik.handleSubmit}>
+        <div className="form-group">
+            <GenericInput
+            label="user name"
+            type="text"
+            name="name"
+            onChange={formik.handleChange}
+            onBlur={formik.handleBlur}
+            value={formik.values.name}
+            width="100%"
+            size="medium"
+          />
+          {formik.touched.name && formik.errors.name ? (
+            <div className="error">{formik.errors.name}</div>
+          ) : null}
+        </div>
+        <div className="form-group">
+             <GenericInput
+            label="email"
+            type="text"
+            name="email"
+            onChange={formik.handleChange}
+            onBlur={formik.handleBlur}
+            value={formik.values.email}
+            width="100%"
+            size="medium"
+          />
+          {formik.touched.email && formik.errors.email ? (
+            <div className="error">{formik.errors.email}</div>
+          ) : null}
+        </div>
+        <div className="form-group" >
+           <GenericInput
+            label="password"
```

Comment:
labels should be taken from constants file

Reply:
קשור לאפרת הרטמן